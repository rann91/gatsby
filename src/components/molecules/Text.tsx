/* @jsx jsx */
import { Container, Divider, Heading, jsx, Text as ThemeText } from 'theme-ui'
import { Strapi_ComponentContentText } from '../../typings/graphql'
import Section from '../atoms/Section'
import TextContainer from '../atoms/TextContainer'

const Text = ({ title, subtitle, text }: Strapi_ComponentContentText) => (
  <Section hasTitle={!!title}>
    <Container px={5}>
      <TextContainer>
        {subtitle && (
          <ThemeText sx={{ variant: 'text.label' }}>{subtitle}</ThemeText>
        )}
        {title && (
          <Heading
            variant="h2"
            sx={{
              '* + &': {
                paddingTop: 2
              }
            }}>
            {title}
          </Heading>
        )}
        {text && (
          <ThemeText
            variant="p"
            as="p"
            sx={{
              '* + &': {
                paddingTop: 6
              }
            }}>
            {text}
          </ThemeText>
        )}
      </TextContainer>
      <Divider />
    </Container>
  </Section>
)

export default Text
