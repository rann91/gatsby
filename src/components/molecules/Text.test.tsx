import { render } from '@testing-library/react'
import React from 'react'
import data from '../../json/text.json'
import Text from './Text'

describe('Text', () => {
  it('renders correctly', () => {
    const { container } = render(<Text {...data} />)

    expect(container).toMatchSnapshot()
  })
})
