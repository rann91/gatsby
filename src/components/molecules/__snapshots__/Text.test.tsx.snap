// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<div>
  <div
    class="css-1ptauik"
    data-testid="section"
  >
    <div
      class="css-1d54mlz"
    >
      <div
        class="css-ycxng9-TextContainer"
        data-testid="text-container"
      >
        <p
          class="css-5n2vnh-Text"
        >
          Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts. Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth.

Even the all-powerful Pointing has no control about the blind texts it is an almost unorthographic li...
        </p>
      </div>
      <hr
        class="css-1q65tqs"
      />
    </div>
  </div>
</div>
`;
